### PR TITLE
Issue1

### PR DESCRIPTION
**Pull request by Allemr**
This pull request returns a JSON list of all albums and the corresponding tracks to the server. The list of albums of is retrieved from the Music folder using a relative address.

To "test" this PR, you'll run the code, change host to localhost, and add `/albumsandtracks` after the port. Verify that you get a JSON list of albums and their tracks (in a list) to the server.

### CODE REVIEW

File: music_server.py [hunk 1]
```diff
@@ -14,6 +16,42 @@ def page_not_found(e):
...
+	"""
+	Retrieve a complete list of albums and the contents
+	of each album on the server.
+	"""
+
+	# Initialize an empty dictionary to store albums and their tracks
+	albums={}
+
+	# Relative path to the Music directory with Albums
+	albums_path = 'Music'
```
Comment (Onekay1331) [inline music_server.py]: I would rename this variable. In this form, it is not necessarily a path. Its rather just the endpoint of a path to some directory, named 'Music'. Would make slightly more sense to name this var as 'directory_name' or some other alternative. 

Or write the string as, 'add\path\to\your\music\directory\' to advise user to add the necessary path

File: music_server.py [hunk 2]
```diff
@@ -14,6 +16,42 @@ def page_not_found(e):
...
+			# Initialize an empty list of tracks
+			tracks = []
+			# List all tracks in the current album directory
+			for track in os.listdir(album_path):
+				if os.path.isfile(os.path.join(album_path, track)):
+					tracks.append(track)
+
+			# Add album and its tracks to albums dictionary
+			albums[album] = tracks
+
```
Comment (Onekay1331) [inline music_server.py:54]: Your loop looks really good. Logic seems to all make sense. I would definitely add some sort of check-handle case here in your root for loop. You have this initial ' if os.path.isdir(album_path): ' statement, but there is no else-statement that handles the case of the statement not being satisfied. 

Or rather, reverse the logic. Could write it as, 'if not os.path.isdir(album_path): ' and within this block, handle the case where the path is invalid or presents some error. And from here you could output, "Some error with this directory" etc. And in the case it passes the if statement, can continue on in the else portion. 

Overall looks great. Definitely needs the error checking and handling functionality added though.

**Review by Onekay1331 — COMMENTED** [review]
Ryan, everything looks to be on a good track. I made a few remarks about some important missing logic and cases. I would check out my comments and lmk what you think! 

Reviewing yours actually reminded me of things I need to check in my own issue...

**Comment by DavidBerdik** [comment]
I will review once @Onekay1331 is done.

**Review by Onekay1331 — DISMISSED** [review]

**Comment by DavidBerdik** [comment]
@Onekay1331 Since Ryan pushed changes to the branch after you had given approval, you will need to review it again.

**Comment by Allemr** [comment]
This pull request has an updated variable name for 'Music' directory, an added check if there is no 'Music' directory, and shortened route name.

To "test" this PR, you'll run the code, change host to localhost, and add `/allalbums` after the port. Test that you get the appropriate message if you do NOT have a folder named 'Music' in your 'Project' directory.  Also, verify that you get an empty dictionary if you have no albums in the 'Music' folder and an empty list in an album with no tracks.

**Review by Onekay1331 — DISMISSED** [review]
lgtm

**Review by DavidBerdik — DISMISSED** [review]
Looks good! Have fun refactoring!

**Review by DavidBerdik — APPROVED** [review]
Looks good to me!

**Pull request merged**